backend/app/routes/admin/putAdmin.js

```diff
@@ -5,32 +5,37 @@ const constants = require('../../../constants');
 
 module.exports = async (req, res, next) => {
   const { firstName, lastName, contact, username } = req.body;
-  const userRecord = await Admin.findOne({
-    email: res.locals.decode.email,
-  });
-
-  if (username === userRecord.username) {
-    return res.status(400).send({ error: 'Username already exists' });
-  }
 
-  const [err] = await to(
-    Admin.findOneAndUpdate({ email: userRecord.email }, { $set: { firstName, lastName, contact, username } })
-  );
+  Admin.findOne({ username }, async (error, userInfo) => {
```

Comment:
This logic will work but this has performance issues. You are using 3 DB calls and DB calls are quite expensive, they contribute a lot to the overall latency. Moreover, the error handling is not proper.
I have reiterated several times that the entire function must use only 1 DB call, please look for optimisation